### ISSUE TITLE
Disconnect another client from server

### ISSUE THREAD

**Issue by dchubad** [nodejs/help#2989]
* **Node.js Version**: node-v12.18.4-x64
* **OS**: Windows 10
* **Scope (install, code, runtime, meta, other?)**: Other
* **Module (and version) (if relevant)**: Unknown

This is my first time using node.js but I have built a working chatroom, however I am at the point of creating commands which are typed into chat for disconnecting other connected clients. If I type in chat /drop clientname I want it to disconnect the clientname from the server (serverside). I can disconnect myself using this method but not another connected client. Below is the function I am using, any help would be much appreciated.

function doDrop(ws, message, temp) {
	ws.personName = message.data;
	s.clients.forEach(function e(client) {
		client.send(JSON.stringify(
			{
				name: "Server",
				data: "🔴 &lt;<b>"+temp[1]+"</b>&gt; has been dropped from the room!"
			}
		));
	});
	var user = temp[1];
	ws.disconnect(user);
return;
};

I have tried disconnect(), close(), Disconnect(), Close() but nothing works. thank

**Comment by RaisinTen** [comment]
Can you please share the rest of the code?

**Comment by dchubad** [comment]
I solved this a long time ago. I have attached my server script but before it will work you will need to open your shell window. Instal all of the following...

npm install -g nodemon
npm install nconf (to store local configuration data)
npm install replaceall
npm install dompurify: installs DOMpurify, like htmlentities server side
npm install jsdom: installs libs for DOMPurify

Cchatroom server script.
[chatroom.zip](https://github.com/nodejs/help/files/5742139/chatroom.zip)

The above is only serverside but you might need to see the clientside scripts to see how it really works. I am sure someone could tidy this script up to make it more efficient, more secure and make it free for everyone to download and use. If you require the clientside scripts let me know. Hope this helps
